package/kernel/linux/modules/lib.mk

```diff
@@ -98,6 +98,132 @@ endef
 $(eval $(call KernelPackage,lib-crc32c))
 
 
+define KernelPackage/lib-crypto-blake2s
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=BLAKE2s hash function library
+  KCONFIG:=CONFIG_CRYPTO_LIB_BLAKE2S
+  FILES:= \
+	$(LINUX_DIR)/lib/crypto/libblake2s.ko \
+	$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
+endef
+
+define KernelPackage/lib-crypto-blake2s/x86/64
+  KCONFIG+=CONFIG_CRYPTO_BLAKE2S_X86
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/blake2s-x86_64.ko
+endef
+
+$(eval $(call KernelPackage,lib-crypto-blake2s))
+
+
+define KernelPackage/lib-crypto-chacha20
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=ChaCha library interface
+  KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA
+  FILES:=$(LINUX_DIR)/lib/crypto/libchacha.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/x86/64
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_X86_64
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/chacha-x86_64.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/arm
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_NEON
+  FILES+=$(LINUX_DIR)/arch/arm/crypto/chacha-neon.ko
+endef
+
+KernelPackage/lib-crypto-chacha20/imx6=$(KernelPackage/lib-crypto-chacha20/arm)
+KernelPackage/lib-crypto-chacha20/ipq40xx=$(KernelPackage/lib-crypto-chacha20/arm)
+KernelPackage/lib-crypto-chacha20/mvebu/cortexa9=$(KernelPackage/lib-crypto-chacha20/arm)
```

Comment:
From grepping elsewhere, it looks like the `/tegra` is a variant that handles vanilla arm (without neon).

Reply:
right, I think @blocktrron added kirkwood so I took that from his branch